packages/next/pages/_app.tsx

```diff
@@ -63,6 +68,16 @@ export default class App<P = {}, CP = P> extends React.Component<
 export class Container extends React.Component {
   componentDidMount() {
     this.scrollToHash()
+
+    // @ts-ignore __NEXT_DATA__ is global
+    if (__NEXT_DATA__.nextExport) {
```

Comment:
What if there's __NEXT_DATA__.query (not sure what the exact naming is), but query coming from the pre-rendered html. Should it be merged? Should we prefer that one? Just making sure that we don't break things.